vars/npmExecuteTests.groovy

```diff
@@ -0,0 +1,38 @@
+import com.sap.piper.ConfigurationHelper
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field String METADATA_FILE = 'metadata/npmExecuteTests.yaml'
+
+@Field Set GENERAL_CONFIG_KEYS = []
+
+@Field Set STEP_CONFIG_KEYS = [
+    "wdi5",
+    "credentialsId",
+]
+
+@Field Set PARAMETER_KEYS = []
+
+void call(Map parameters = [:]) {
+    final script = checkScript(this, parameters) ?: this
+    String stageName = parameters.stageName ?: env.STAGE_NAME
+    Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults([:], stageName)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, stageName, PARAMETER_KEYS)
+            .mixin(parameters, STEP_CONFIG_KEYS)
+            .use()
+
+    List credentials = []
+    if (config.credentialsId) {
+        if (config.wdi5) {
+            credentials.add([type: 'usernamePassword', id: config.credentialsId, env: ['wdi5_username', 'wdi5_password']])
+        } else {
+            credentials.add([type: 'usernamePassword', id: config.credentialsId, env: ['e2e_username', 'e2e_password']])
+        }
+    }
```

Comment:
We should avoid doing such special handling in the groovy. That way we will have no support for it in GHA / ADO.

Reply:
So should we store those in Vault as well?

Reply:
Something like:
```
{
  "appSecrets": {
  	"urls": [
      {
      	"url": "loocalhost:8080",
        "username": "some-username1",
        "password": "some-password1"
      },
      {
      	"url": "loocalhost:8080",
        "username": "some-username2",
        "password": "some-password2"
      }
    ],
    "username": "",
    "password": ""
  }
}
```